lib/_http_client.js

```diff
@@ -336,7 +336,16 @@ function ClientRequest(input, options, cb) {
     // No agent, default to Connection:close.
     this._last = true;
     this.shouldKeepAlive = false;
-    if (typeof optsWithoutSignal.createConnection === 'function') {
+    let opts = optsWithoutSignal;
+    if (opts.path || opts.socketPath) {
+      opts = { ...optsWithoutSignal };
+      if (opts.socketPath) {
+        opts.path = opts.socketPath;
+      } else if (opts.path) {
+        delete opts.path;
```

Comment:
Nit: I would prefer `opts.path = undefined` so we avoid to deoptimize the object. If it doesn't interfere with the rest of the functionality.

Reply:
Done, Thanks!